eclipse-collections/src/main/java/org/eclipse/collections/impl/map/AbstractSynchronizedMapIterable.java

```diff
@@ -350,15 +349,22 @@ public V updateValue(K key, Function0<? extends V> factory, Function<? super V,
         }
     }
 
+    @Override
+    public RichIterable<K> keysView()
+    {
+        return LazyIterate.adapt(this.keySet());
+    }
+
+    @Override
+    public RichIterable<V> valuesView()
+    {
+        return LazyIterate.adapt(this.values());
+    }
+
     @Override
     public RichIterable<Pair<K, V>> keyValuesView()
     {
-        synchronized (this.lock)
-        {
-            Set<Entry<K, V>> entries = this.getDelegate().entrySet();
-            Iterable<Pair<K, V>> pairs = Iterate.collect(entries, AbstractImmutableEntry.getPairFunction());
-            return LazyIterate.adapt(pairs);
-        }
+        return LazyIterate.adapt(this.entrySet()).collect(AbstractImmutableEntry.getPairFunction());
```

Comment:
This implementation was wrong. By delegating to a non-lazy implementation of collect(), it caused observable changes in iteration order on collections that have a sort order using a comparator.

The test MapIterableTestCase#Iterable_toString() catches the bug when run in SynchronizedSortedMapTest. The assertion:

```java
Assert.assertEquals(
        "[10:4, 9:4, 8:4, 7:4, 6:3, 5:3, 4:3, 3:2, 2:2, 1:1]",
        this.newWith(4, 4, 4, 4, 3, 3, 3, 2, 2, 1).keyValuesView().toString());
```

Fails with:

```
Expected :[10:4, 9:4, 8:4, 7:4, 6:3, 5:3, 4:3, 3:2, 2:2, 1:1]
Actual   :[4:3, 7:4, 8:4, 10:4, 9:4, 6:3, 5:3, 3:2, 2:2, 1:1]
```